### DISCUSSION TITLE
how can we specify some module in one lib to bundle

### DISCUSSION THREAD

**Discussion by wanghaisheng** [marcelotduarte/cx_Freeze#2166] [Q&A]
lets say numpy, it got 70M

if we only want "numpy.linspace" releated module to bundle into my exe

**Comment by marcelotduarte** [comment]
Use [includes](https://cx-freeze.readthedocs.io/en/stable/setup_script.html#cmdoption-arg-includes) and [excludes](https://cx-freeze.readthedocs.io/en/stable/setup_script.html#cmdoption-arg-excludes).
But in the case of numpy, I can tell you that you won't be able to exclude many modules, as some depend on each other and we had to make [hooks](https://github.com/marcelotduarte/cx_Freeze/blob/main/cx_Freeze/hooks/numpy.py) to control this.

**Reply by wanghaisheng** [reply]
that is great. I will look at it.
